### PR TITLE
hwdef: add YJUAV_A6SE_H743 board support

### PR DESCRIPTION
**Pull request by yunjiuav**

### CODE REVIEW

**Review by khancyr — APPROVED** [review]
LGTM
Got docs and support is done by maintainer.

**Comment by peterbarker** [comment]
@Hwurzburg likes to review the docs for these boards as they often need patching.

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+PE5 EXT_CS CS
+
+# I2C buses
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+PH7 I2C3_SCL I2C3
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat:99]: where is this pinned out?

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+PH8 I2C3_SDA I2C3
+
+PF14 I2C4_SCL I2C4
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat:102]: where is this pinned out?

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+ - SERIAL4 -> USART2 (GPS2), NODMA
+ - SERIAL5 -> UART8 (SBUS)
+ - SERIAL6 -> UART7 (Debug), NODMA
+ - SERIAL7 -> USB2 (OTG2)
+
+## RC Input
+
+The remote control signal should be connected to the “RC IN” pin, at one side of the servo channels.
+
+This signal pin supports two types of remote control signal inputs, SBUS and PPM signals.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md:161]: this supports all unidirectional protocols...PPM,SBus,DSM,Fport(RC only),SRXL,etc

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+| :--: | :----: | :---: |
+|  1   | VCC_IN |  +5V  |
+|  2   |   DM   | +3.3V |
+|  3   |   DP   | +3.3V |
+|  4   |  GND   |  GND  |
+
+## UART Mapping
+
+ - SERIAL0 -> USB (OTG1)
+ - SERIAL1 -> USART1 (Telem1)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md:149]: this is USART2,not 1

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+ - SERIAL2 -> USART6 (Telem2)
+ - SERIAL3 -> USART3 (GPS1), NODMA
+ - SERIAL4 -> USART2 (GPS2), NODMA
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md:152]: this is USART1, not 2

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/defaults.parm [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# setup the heater temperature to 45 degree
+BRD_HEAT_TARG 45
+
+# turn on the CAN power monitoring(default)
+CAN_P1_DRIVER 1
+BATT_MONITOR 4
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/defaults.parm:6]: should be in hwdef

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/defaults.parm [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+# setup the parameter for the ADC power module
+BATT_VOLT_MULT 21
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/defaults.parm:9]: should be in hwdef, and the following line below this...also ou are missing the pin defs for volt and current monitoring

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+|  2   | CAN_P  | +3.3V |
+|  3   | CAN_N  | +3.3V |
+|  4   |  GND   |  GND  |
+
+**GPS1**
+
+| Pin  | Signal  | Volt  |
+| :--: | :-----: | :---: |
+|  1   |   VCC   |  +5V  |
+|  2   | UART_TX | +3.3V |
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md:94]: should have uart#

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+|  4   | I2C_SCL | +3.3V |
+|  5   | I2C_SDA | +3.3V |
+|  6   |   GND   |  GND  |
+
+**GPS2&SAFETY**
+
+| Pin  |    Signal     | Volt  |
+| :--: | :-----------: | :---: |
+|  1   |      VCC      |  +5V  |
+|  2   |    UART_TX    | +3.3V |
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md:105]: should have uart #

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+|  4   |  GND   |  GND  |
+
+## UART Mapping
+
+ - SERIAL0 -> USB (OTG1)
+ - SERIAL1 -> USART1 (Telem1)
+ - SERIAL2 -> USART6 (Telem2)
+ - SERIAL3 -> USART3 (GPS1), NODMA
+ - SERIAL4 -> USART2 (GPS2), NODMA
+ - SERIAL5 -> UART8 (SBUS)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/README.md:153]: this not setup to output this

**Comment by Hwurzburg** [comment]
> @Hwurzburg likes to review the docs for these boards as they often need patching.

more than just docs often...functional issues...as this one has

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

**Comment by yunjiuav** [comment]
@Hwurzburg Sorry. I have carefully modified README. Please help me check again. thanks

**Review by Hwurzburg — APPROVED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# enable RAMTROM parameter storage
+define HAL_STORAGE_SIZE 32768
+define HAL_WITH_RAMTRON 1
+
+DMA_PRIORITY SPI1*  TIM*UP*
+DMA_NOSHARE  SPI1*  TIM*UP*
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat:240]: I would suggest no share for SPI6 as well

Comment (yunjiuav) [reply libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat:240]: This is a good suggestion.

**Comment by yunjiuav** [comment]
Modification completed, please help me mark the merge. I will submit Wiki for YJUAV_ A6SE_ H743.

**Comment by andyp1per** [comment]
hi @yunjiuav if possible can you squash the commits together?

**Comment by yunjiuav** [comment]
@andyp1per Is that so?

**Comment by andyp1per** [comment]
Looks correct

**Review by andyp1per — APPROVED** [review]

**Pull request merged**